cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -2317,6 +2372,21 @@ const FunctionDoc utf8_lower_doc(
     "Transform input to lowercase",
     ("For each string in `strings`, return a lowercase version."), {"strings"});
 
+const FunctionDoc ascii_reverse_doc(
+    "Reverse ASCII input",
+    ("For each ASCII string in `strings`, return a reversed version.\n\n"
+     "This function assumes the input is fully ASCII.  If it may contain\n"
+     "non-ASCII characters, use \"utf8_reverse\" instead."),
+    {"strings"});
+
+const FunctionDoc utf8_reverse_doc(
+    "Reverse utf8 input",
+    ("For each utf8 string in `strings`, return a reversed version.\n\n"
+     "This function works on UTF8 codepoint level. As a consequence, if the input \n"
+     "contains combining UTF8 character sequences/ 'graphemes', they would also \n"
+     "be reversed."),
```

Comment:
Just to reword things a bit for clarity perhaps:

> Reverse input
> For each string in `strings`, return a reversed version.
> This function operates on codepoints/UTF-8 code units, not grapheme clusters. Hence, it will not correctly reverse grapheme clusters composed of multiple codepoints.

As we assume strings are UTF-8 and don't need to repeat that.